lib/livebook_web/live/session_live.ex

```diff
@@ -1703,6 +1703,14 @@ defmodule LivebookWeb.SessionLive do
     end
   end
 
+  defp after_operation(
+         socket,
+         _prev_socket,
+         {:add_cell_evaluation_output, _client_id, cell_id, {:doctest_result, result}}
+       ) do
+    push_doctest_event(socket, cell_id, result)
```

Comment:
We can send the same event and decide what to do with it on the client

```suggestion
    push_event(socket, "doctest_result:#{cell_id}", %{state: result.state, line: doctest_line})
```